### PR TITLE
Adding PXT link command to the pxt cli

### PR DESCRIPTION
**Pull request by riknoll**
We'll have to publish this manually since it's in the JS cli and not `cli.ts`

### CODE REVIEW

File: pxt-cli/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pxt",
-  "version": "0.3.0",
+  "version": "0.3.1",
```
Comment (mmoskal) [inline pxt-cli/package.json]: Should be 0.4.0

File: pxt-cli/cli.js [hunk 2]
```diff
@@ -70,13 +70,60 @@ function target(n) {
...
+        console.log("Could not find " + absPath);
+        process.exit(1);
+    }
+
+    var modulesPath = path.resolve("node_modules");
+
+    if (!fs.existsSync(modulesPath))
+        fs.mkdirSync(modulesPath)
+
+    var corePath = path.join(modulesPath, "pxt-core");
```
Comment (mmoskal) [inline pxt-cli/cli.js]: What about `pxt-common-packages` ? You could look in `dir/package.json` for `name`

File: pxt-cli/cli.js [hunk 3]
```diff
@@ -70,13 +70,60 @@ function target(n) {
...
+    if (fs.existsSync(corePath)) {
+        var stats = fs.statSync(corePath);
+        if (stats.isSymbolicLink()) {
+            fs.unlinkSync(corePath);
+        }
+        else {
+            try {
+                var rimraf = require("rimraf");
```
Comment (mmoskal) [inline pxt-cli/cli.js]: The package doesn't require rimraf, but maybe this is fine

Comment (riknoll) [reply pxt-cli/cli.js]: Yeah, but pxt-core does

**Comment by riknoll** [comment]
If we put it in cli.ts then we'll have to maintain it in V0 and master. What's the issue with keeping it in the JS?

File: pxt-cli/cli.js (hunk 2 above)
Comment (riknoll) [reply pxt-cli/cli.js]: That's a great idea

**Comment by mmoskal** [comment]
OK, let's put it here. But it would be nice to support pxt-common-packages as well.

**Comment by riknoll** [comment]
I'll add common-packages support

**Comment by riknoll** [comment]
Done! @mmoskal take a look

**Comment by mmoskal** [comment]
LGTM!

**Pull request merged**